### DISCUSSION TITLE
Simulating M3 BLQ Models

### DISCUSSION THREAD

**Discussion by ldbro0** [metrumresearchgroup/mrgsolve#886] [Q&A]
Hi Kyle, I'm interested in implementing a nonmem model using the M3 method, is this something supported by mrgsolve?

**Comment by kylebaron** [comment]
Hi @ldbro0 - 

What aspect of your M3 model were you hoping to capture through simulation?  IMO, there isn't much to do for simulating from M3 models with `mrgsolve`.  `mrgsolve` will just simulate data for you and that data is either above or below the quantitation limit. So if you're simulating data you can truncate it at the quantitation limit or maybe replace values below the quantitation limit with `NA_real_` or something like that. But in the end you have control over that after the simulation is finished. 

Let me know if you wanted something else out of the simulation.

Thanks,
Kyle

**Reply by ldbro0** [reply]
That makes sense!  Thanks for clarifying!

**Reply by kylebaron** [reply]
Np; good luck with the simulation. 
Kyle
